### ISSUE TITLE
Installing via Composer global require leaves me without packages directory

### ISSUE THREAD

**Issue by monolux** [wp-cli/wp-cli#4144]
Installing wp-cli via [composer global require](https://make.wordpress.org/cli/handbook/installing/#installing-via-composer) with command `composer global require wp-cli/wp-cli` gives me OK output like so.

```
D:\>composer global require wp-cli/wp-cli
Changed current directory to C:/Users/User/AppData/Roaming/Composer
Using version ^1.2 for wp-cli/wp-cli
./composer.json has been created
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 51 installs, 0 updates, 0 removals
  - Installing wp-cli/autoload-splitter (v0.1.3): Downloading (100%)
  - Installing wp-cli/php-cli-tools (v0.11.2): Downloading (100%)
  - Installing wp-cli/mustangostang-spyc (0.6.3): Downloading (100%)
  - Installing composer/semver (1.4.2): Downloading (100%)
  - Installing mustache/mustache (v2.11.1): Downloading (100%)
  - Installing ramsey/array_column (1.1.3): Downloading (100%)
  - Installing rmccue/requests (v1.7.0): Downloading (100%)
  - Installing symfony/filesystem (v3.3.2): Downloading (100%)
  - Installing symfony/config (v3.3.2): Downloading (100%)
  - Installing psr/log (1.0.2): Downloading (100%)
  - Installing symfony/debug (v3.3.2): Downloading (100%)
  - Installing symfony/polyfill-mbstring (v1.3.0): Downloading (100%)
  - Installing symfony/console (v3.3.2): Downloading (100%)
  - Installing psr/container (1.0.0): Downloading (100%)
  - Installing symfony/dependency-injection (v3.3.2): Downloading (100%)
  - Installing symfony/event-dispatcher (v3.3.2): Downloading (100%)
  - Installing symfony/finder (v3.3.2): Downloading (100%)
  - Installing symfony/process (v3.3.2): Downloading (100%)
  - Installing symfony/translation (v3.3.2): Downloading (100%)
  - Installing symfony/yaml (v3.3.2): Downloading (100%)
  - Installing wp-cli/cache-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/checksum-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/config-command (v1.1.1): Downloading (100%)
  - Installing wp-cli/core-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/cron-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/widget-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/super-admin-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/wp-cli (v1.2.1): Downloading (100%)
    Skipped installation of bin bin/wp.bat proxy for package wp-cli/wp-cli: a .bat proxy was already installed
  - Installing wp-cli/shell-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/server-command (v1.0.5): Downloading (100%)
  - Installing wp-cli/search-replace-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/scaffold-command (v1.0.4): Downloading (100%)
  - Installing wp-cli/role-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/rewrite-command (v1.0.1): Downloading (100%)
  - Installing seld/phar-utils (1.0.1): Downloading (100%)
  - Installing seld/jsonlint (1.6.0): Downloading (100%)
  - Installing seld/cli-prompt (1.0.3): Downloading (100%)
  - Installing justinrainbow/json-schema (5.2.1): Downloading (100%)
  - Installing composer/spdx-licenses (1.1.6): Downloading (100%)
  - Installing composer/ca-bundle (1.0.7): Downloading (100%)
  - Installing composer/composer (1.4.2): Downloading (100%)
  - Installing wp-cli/package-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/media-command (v1.0.2): Downloading (100%)
  - Installing wp-cli/language-command (v1.0.2): Downloading (100%)
  - Installing wp-cli/import-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/extension-command (v1.1.0): Downloading (100%)
  - Installing nb/oxymel (v0.1.0): Downloading (100%)
  - Installing wp-cli/export-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/eval-command (v1.0.1): Downloading (100%)
  - Installing wp-cli/entity-command (v1.0.2): Downloading (100%)
  - Installing wp-cli/db-command (v1.1.0): Downloading (100%)
symfony/dependency-injection suggests installing symfony/expression-language (For using expressions in service container
 configuration)
symfony/dependency-injection suggests installing symfony/proxy-manager-bridge (Generate service proxies to lazy load the
m)
symfony/event-dispatcher suggests installing symfony/http-kernel ()
wp-cli/wp-cli suggests installing psy/psysh (Enhanced `wp shell` functionality)
Writing lock file
Generating autoload files
D:\>
```

Checking wp-cli install with `wp --info` however leaves me without wp-cli packages dir.
```

D:\>wp --info
PHP binary:     D:\XAMPP\php\php.exe
PHP version:    5.6.8
php.ini used:   D:\XAMPP\php\php.ini
WP-CLI root dir:        phar://wp-cli.phar
WP-CLI packages dir:
WP-CLI global config:
WP-CLI project config:
WP-CLI version: 1.1.0
```

How can I install wp-cli globally via composer on Windows so that I do indeed get the WP-CLI packages dir set up please?

**Comment by monolux** [comment]
I am referencing this issue https://github.com/wp-cli/wp-cli/issues/4145 since that is a direct outcome of wp-cli installation via Composer. If it is not possible to get the WP-CLI packages dir automatically set up during installation with Composer I would like to know if it is possible to set the WP-CLI packages dir manually. Hence I created the second issue.

**Comment by monolux** [comment]
Unfortunately Compose **does not** set the wp-cli environment variable automatically. This is not possible or won't be done with the installation of wp-cli through Composer.

However i **can** be done manually like so.

Due to help from @schlessera and @danielhuesken on the #cli WordPress Slack channel I found out what I had to do.

Instead of setting a PATH value in the Windows environment variables I had to make a **new** variable called `WP_CLI_PACKAGES_DIR` and set the value of that variable to `C:\Users\YOUR_WINDOWS_YOURSERNAME_HERE\.wp-cli\packages`.

I also had to create the folder `packages` under `C:\Users\YOUR_WINDOWS_YOURSERNAME_HERE\.wp-cli\`.  The folder was not created automatically.

After that though Composer was complaining that PHP does not have enough memory and so I permanently increased the memory from `128M` to `512M` in the php.ini file.

Now I could install the wp-cli package just fine, ran it and it also worked perfectly.

Reference links:
https://superuser.com/questions/949560/how-do-i-set-system-environment-variables-in-windows-10

https://make.wordpress.org/cli/handbook/common-issues/#php-fatal-error-allowed-memory-size-of-999999-bytes-exhausted-tried-to-allocate-99-bytes
